pumps/mongo.go

```diff
@@ -103,6 +107,115 @@ func (m *MongoPump) Init(config interface{}) error {
 	return nil
 }
 
+func (m *MongoPump) capCollection() {
+
+	var colName = m.dbConf.CollectionName
+	var colCapMaxSizeBytes = m.dbConf.CollectionCapMaxSizeBytes
+
+	exists, err := m.collectionExists(colName)
+	if err != nil {
+		// TODO Handle Error
+
+		return
+	}
+
+	if !exists {
+		if err := m.createCappedCollection(colName, colCapMaxSizeBytes); err != nil {
+			// TODO handle error
+
+			return
+		}
+
+		return
+	}
+
+	// If colCapMaxSizeBytes == 0, uncap the collection and return
+	if colCapMaxSizeBytes == 0 {
+		m.uncapCollection(colName)
+
+		return
+	}
+
+	m.resizeCappedCollection(colName, colCapMaxSizeBytes)
+}
+
+// collectionExists checks to see if a collection name exists in the db.
+func (m *MongoPump) collectionExists(name string) (bool, error) {
+	sess := m.dbSession.Copy()
+	defer sess.Close()
+
+	colNames, err := sess.DB("").CollectionNames()
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"prefix": mongoPrefix,
+		}).Error("Unable to get column names: ", err)
+
+		return false, err
+	}
+
+	for _, coll := range colNames {
+		if coll == name {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (m *MongoPump) createCappedCollection(name string, maxBytes int) error {
+	sess := m.dbSession.Copy()
+	defer sess.Close()
+
+	// No need to create collection if MaxBytes are 0
+	if maxBytes == 0 {
+		return nil
+	}
+
+	return sess.DB("").C(name).Create(&mgo.CollectionInfo{Capped: true, MaxBytes: maxBytes})
+}
+
+func (m *MongoPump) uncapCollection(name string) {
```

Comment:
Make sure this is possible in older versions of MongoDB, not everyone runs latest.